zirgen/Dialect/Zll/IR/Interpreter.cpp

```diff
@@ -44,7 +44,8 @@ template <typename T> void formatFieldElem(const InterpVal* interpVal, llvm::raw
 std::vector<uint64_t> ExternHandler::doExtern(llvm::StringRef name,
                                               llvm::StringRef extra,
                                               llvm::ArrayRef<const InterpVal*> args,
-                                              size_t outCount) {
+                                              size_t outCount,
+                                              bool* failed) {
```

Comment:
The use of pointer-to-bool in this API leaves me feeling sketchy, both because of the vagueness about lifetimes associated with bare pointers, and because the caller can easily (perhaps inadvertently) ignore the failure of a call. Would it work to return `std::optional<std::vector<uint64_t>>` instead, using the absence of a result to indicate failure? Or, if that's too heavy, perhaps use a `bool&` instead?

Perhaps these ideas have been considered and rejected - happy to accept this if there's reason to do it this way - but it'd be good to know.

Reply:
I was reluctant because there were a lot of places where existing extern handlers use `return {};`, which would then mean `std::nullopt` instead of an empty array. I got over it and fixed all of these, because you're right that it's better to do it that way.